core/src/main/python/synapse/ml/core/utils/platform.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in project root for information.
+import os
+from pyspark.sql import SparkSession
+
+
+def current_platform():
+    if os.environ.get("AZURE_SERVICE", None) == "Microsoft.ProjectArcadia":
+        return "synapse"
+    elif "dbfs" in os.listdir("/"):
+        return "databricks"
+    elif os.environ.get("BINDER_LAUNCH_HOST", None) is not None:
+        return "binder"
+    else:
+        return "unknown"
+
```

Comment:
We might also want to add a findSecret API that will run through the gambit of looking in keyvaults, using databricks methods, and checking env vars before yelling and asking someone to replace that line with their key if they cant find anything